### PR TITLE
refactor: Convert youth risk assessment to a class

### PR DESCRIPTION
**Pull request by Thource**
### What changed

The youth risk assessment service has been converted to a class extending BaseService.

### Why did it change

This change was necessary so that we can pass request through to the api client in the future.

### Issue tracking

- [P4-1776](https://dsdmoj.atlassian.net/browse/P4-1776)

## Screenshots

| Before | After |
| ------ | ----- |
| <kbd></kbd> | <kbd></kbd> |

#### Automated testing

- [x] Added unit tests to cover changes
- [ ] Added end-to-end tests to cover any journey changes

#### Manual testing

Has been tested in the following browsers:

- [ ] Chrome
- [ ] Firefox
- [ ] Edge
- [ ] Internet Explorer

### Environment variables

- [x] No environment variables were added or changed

- [ ] Documented in the [README](ministryofjustice/hmpps-book-secure-move-frontend/blob/master/README.md)
- [ ] Added to [continous integration](https://app.circleci.com/pipelines/github/ministryofjustice/hmpps-book-secure-move-frontend)
- [ ] Added to staging environment (deployment repository)
- [ ] Added to preproduction environment (deployment repository)
- [ ] Added to production environment (deployment repository)

### Other considerations

Commit messages with a `fix` or `feat` type are automatically used to generate the [changelog](ministryofjustice/hmpps-book-secure-move-frontend/blob/master/CHANGELOG.md) and
[GitHub release notes](https://github.com/ministryofjustice/hmpps-book-secure-move-frontend/releases) during the release task. Please make sure they will read well on their own in a
summary of changes and that the commit body gives a more detailed description of those changes if necessary.

- [x] No new [Personally Identifiable Information (PII)](https://support.google.com/analytics/answer/6366371?hl=en) is being sent via analytics
- [ ] Update [README](ministryofjustice/hmpps-book-secure-move-frontend/blob/master/README.md) with any new instructions or tasks

### CODE REVIEW

File: common/controllers/framework/framework-step.js [hunk 1]
```diff
@@ -2,10 +2,11 @@ const { isEmpty, fromPairs, snakeCase } = require('lodash')
 
 const fieldHelpers = require('../../helpers/field')
 const frameworksHelpers = require('../../helpers/frameworks')
+const YouthRiskAssessmentService = require('../../services/youth-risk-assessment')
 const FormWizardController = require('../form-wizard')
 const services = {
   'person-escort-record': require('../../services/person-escort-record'),
-  'youth-risk-assessment': require('../../services/youth-risk-assessment'),
+  'youth-risk-assessment': new YouthRiskAssessmentService(),
```
Comment (Nimphal) [inline common/controllers/framework/framework-step.js:9]: Ooooh, we probably want to move where this is instantiated to be within the controller somehow, so it's instantiated with the req. Perhaps services can be set in a constructor method in the controller, which first calls super() and then sets this.services

File: common/services/youth-risk-assessment.test.js [hunk 2]
```diff
@@ -3,13 +3,14 @@ const proxyquire = require('proxyquire')
...
 const mockFrameworksVersion = '2.5.3'
-const youthRiskAssessmentService = proxyquire('./youth-risk-assessment', {
+const YouthRiskAssessmentService = proxyquire('./youth-risk-assessment', {
   '../../config': {
     FRAMEWORKS: {
       CURRENT_VERSION: mockFrameworksVersion,
     },
   },
 })
+const youthRiskAssessmentService = new YouthRiskAssessmentService()
```
Comment (Nimphal) [inline common/services/youth-risk-assessment.test.js:13]: this should probably have the { apiClient } in :)

File: common/controllers/framework/framework-step.js [hunk 3]
```diff
@@ -134,10 +129,15 @@ class FrameworkStepController extends FormWizardController {
...
       if (responses.length) {
-        await services[assessment.framework.name].respond(
-          assessment.id,
-          responses
+        // convert "service-name" to "serviceName"
+        const serviceName = assessment.framework.name.replace(
+          /-([a-z])/g,
+          function (g) {
+            return g[1].toUpperCase()
+          }
```
Comment (Nimphal) [inline common/controllers/framework/framework-step.js:137]: oh wow, regex and everything! If this is the only place this is used, you could just move the mapping from the previous version within this function. It's a bit clunkier, but a bit more explicit about what's happening. So something like:
```
const services = {
  'person-escort-record': req.services.personEscortRecord,
  'youth-risk-assessment': req.services.youthRiskAssessment,
}
```

File: common/controllers/framework/framework-step.test.js [hunk 4]
```diff
@@ -1,11 +1,14 @@
...
 const frameworksHelpers = require('../../../common/helpers/frameworks')
+
+const youthRiskAssessmentService = {
+  respond: sinon.stub(),
+}
 const services = {
-  'person-escort-record': require('../../services/person-escort-record'),
-  'youth-risk-assessment': require('../../services/youth-risk-assessment'),
+  personEscortRecord: require('../../services/person-escort-record'),
+  youthRiskAssessment: youthRiskAssessmentService,
```
Comment (Nimphal) [inline common/controllers/framework/framework-step.test.js:10]: probably want to stub the person escort one here as well

Comment (Thource) [reply common/controllers/framework/framework-step.test.js:10]: I was going to leave this for when the PER service gets converted to a class :)

Comment (Nimphal) [inline common/controllers/framework/framework-step.test.js:10]: These probably want to match the controller now.

Comment (Thource) [reply common/controllers/framework/framework-step.test.js:10]: I think this is correct, this `services` mocks `req.services`

Comment (teneightfive) [reply common/controllers/framework/framework-step.test.js:10]: I wonder if this whole thing can be simplified and not need to fake importing them?

And create the stub on `req.services` directly?

Comment (Nimphal) [reply common/controllers/framework/framework-step.test.js:10]: Yeah, I'll look at different approaches in my next PR, there's a couple of places we do this.

**Review by Nimphal — APPROVED** [review]

**Pull request merged**